core/swupdate.c

```diff
@@ -772,9 +772,10 @@ int main(int argc, char **argv)
 			break;
 #ifdef CONFIG_DOWNLOAD
 		case 'd':
-			(void)snprintf(dwloptions, sizeof(dwloptions), "%s %s", argv[0], optarg);
```

Comment:
@TomaszAIR It looks like we silently assume that the memory for string will always be correctly allocated.
What if the allocation fails? We should handle that case and break the program execution with error informing user what's going on. The commit suggested  by Stefano hanldes errors in some way: e4756a67336b48150c576adda1c8faca5cfb9294

Please note that those options are directly entered by the user and we must expect anything there and handle in correctly.